routing/turns_generator.hpp

```diff
@@ -29,36 +29,54 @@ namespace turns
 using TGetIndexFunction = function<size_t(pair<size_t, size_t>)>;
 
 /*!
- * \brief The TurnInfo struct is an accumulator for all junction information.
- * During a junction (a turn) analysis a different subsets of fields in the structure
- * may be calculated. The main purpose of TurnInfo is to prevent recalculation the same fields.
- * The idea is if a field is required to check whether a field has been calculated.
- * If yes, just use it. If not, the field should be calculated, kept in the structure
- * and used.
+ * \brief The LoadedPathSegment struct is a representation for a single osrm node path.
+ * It unpacks all necessary information about path and stores it inside this structure.
+ * Postprocessing must read information from the structure and does not initiate disk readings.
  */
-struct TurnInfo
+struct LoadedPathSegment
 {
-  RoutingMapping & m_routeMapping;
-
-  NodeID m_ingoingNodeID;
-  OsrmMappingTypes::FtSeg m_ingoingSegment;
-  ftypes::HighwayClass m_ingoingHighwayClass;
-  bool m_isIngoingEdgeRoundabout;
+  vector<m2::PointD> m_path;
+  ftypes::HighwayClass m_highwayClass;
+  bool m_onRoundabout;
+  bool m_isLink;
+  EdgeWeight m_weight;
+  string m_name;
+  NodeID m_nodeId;
+  vector<SingleLaneInfo> m_lanes;
+
+  // General constructor.
+  LoadedPathSegment(RoutingMapping & mapping, Index const & index,
+                    RawPathData const & osrmPathSegment);
+  // Spesial constructor for side nodes. Splits OSRM node by information from the FeatureGraphNode.
+  LoadedPathSegment(RoutingMapping & mapping, Index const & index,
+                    RawPathData const & osrmPathSegment, FeatureGraphNode const & startGraphNode,
+                    FeatureGraphNode const & endGraphNode, bool isStartNode, bool isEndNode);
+  LoadedPathSegment() = delete;
+
+private:
+  void LoadPathGeometry(buffer_vector<OsrmMappingTypes::FtSeg, 8> const & buffer, size_t startK,
+                        size_t endK, Index const & index, RoutingMapping & mapping,
```

Comment:
Please, add a comment to this method, and rename startK/endK to something meaningful.